core/imt/src/TThreadExecutor.cxx

```diff
@@ -75,6 +75,32 @@
 ///
 //////////////////////////////////////////////////////////////////////////
 
+namespace ROOT {
+namespace Internal {
+
+/// A helper function to implement the TThreadExecutor::ParallelReduce methods
+template<typename T>
+static T ReduceHelper(const std::vector<T> &objs, const std::function<T(T a, T b)> &redfunc)
+{
+   using BRange_t = tbb::blocked_range<decltype(objs.begin())>;
+
+   auto pred = [redfunc](BRange_t const & range, T init) {
+      return std::accumulate(range.begin(), range.end(), init, redfunc);
+   };
+
+   T result;
+
+   BRange_t objRange(objs.begin(), objs.end());
+
+   tbb::this_task_arena::isolate([&]{
+      result = tbb::parallel_reduce(objRange, T{}, pred, redfunc);
+   });
```

Comment:
`isolate` returns the same type that the enclosing lambda returns, so this could be

```suggestion
const auto result = bb::this_task_arena::isolate([&]{
      return tbb::parallel_reduce(objRange(objs.begin(), objs.end()), T{}, pred, redfunc);
});
```

Reply:
Doesn't this basically stop all work stealing in ROOT? How does this affect performance?

Reply:
no it doesn't, see the last 4 paragraphs [here](https://software.intel.com/en-us/node/684814) and the actual method reference [here](https://software.intel.com/en-us/node/506359)

about the performance impact...only way is to measure

Reply:
Oh ok, thanks!

Reply:
we measured performance and the results were good. A table with the findings will be added below.